src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -1882,6 +1882,31 @@ public void testContainerWithMoreCpuOptions() throws Exception {
   }
 
   @Test
+  public void testContainerWithNanoCpusOptions() throws Exception {
```

Comment:
There are some good examples in DefaultDockerClientUnitTest but the idea is that you can have some sample JSON files that represent the different kinds of responses the actual docker daemon might return.  I think its harder to test certain scenarios if you are using a live docker daemon (also the tests run slower).